tasks/fullStack/app/jobs/create_review_worker.rb

```diff
@@ -0,0 +1,31 @@
+class CreateReviewWorker
+  include Sidekiq::Worker
+
+  def perform(shop_id, product_id, body, rating, reviewer_name, tags)
+    Review.transaction do
+      review = Review.create!(product_id: product_id, body: body, rating: rating, reviewer_name: reviewer_name, tags: tags)
+      monthly_avg_rating = MonthlyAvgRating.where(shop_id: shop_id, start_date: review.created_at.beginning_of_month).first
+      if monthly_avg_rating.present?
+        MonthlyAvgRating.connection.execute <<-SQL
+          UPDATE monthly_avg_ratings
+          SET total_ratings = total_ratings + #{rating}, total_reviews = total_reviews + 1
```

Comment:
`rating` is raw user input. So would this cause SQL injection?

Reply:
But interesting approach in incrementing rating and count in an "atomic" query.

Reply:
No, I validated the input before passing it to the creation worker, so the `rating` is confirmed to be a number.